### PR TITLE
A11-3052 Add helpfully disabled tooltip

### PR DESCRIPTION
**Pull request by charbelrami**
# :wrench: Modifying a component

## Context

Adds `helpfullyDisabled` option to the Tooltip's API. The behavior is the same as `auxiliaryDescription`.

[A11-3052 : Add a new Tooltip type name like Tooltip.helpfullyDisabled](https://linear.app/noredink/issue/A11-3052/add-a-new-tooltip-type-name-like-tooltiphelpfullydisabled)

## :framed_picture: What does this change look like?

helpfully disabled tooltip docs

## Component completion checklist

- [x] I've gone through the relevant sections of the [Development Accessibility guide](https://paper.dropbox.com/doc/Accessibility-guide-4-Development--BiIVdijSaoijjOuhz3iTCJJ1Ag-rGoHpC91pFg3zTrYpvOCQ) with the changes I made to this component in mind
- [x] Changes are clearly documented
    - [x] Component docs include a changelog
    - [x] Any new exposed functions or properties have docs
- [x] Changes extend to the Component Catalog
    - [x] The Component Catalog is updated to use the newest version, if appropriate
    - [x] The Component Catalog example version number is updated, if appropriate
    - [x] Any new customizations are available from the Component Catalog
    - [x] The component example still has:
        - an accurate preview
        - valid sample code
        - correct keyboard behavior
        - correct and comprehensive guidance around how to use the component
- [x] Changes to the component are tested/the new version of the component is tested
- [x] Component API follows standard patterns in noredink-ui
    - e.g., as a dev, I can conveniently add an `nriDescription`
    - and adding a new feature to the component will _not_ require major API changes to the component
- [x] If this is a new major version of the component, our team has stories created to upgrade all instances of the old component. Here are links to the stories:
  - add your story links here OR just write this is not a new major version

### CODE REVIEW

File: src/Nri/Ui/Tooltip/V3.elm [hunk 1]
```diff
@@ -835,6 +836,12 @@ auxiliaryDescription =
     Attribute (\config -> { config | purpose = AuxillaryDescription })
 
 
+{-| -}
+helpfullyDisabled : Attribute msg
+helpfullyDisabled =
+    Attribute (\config -> { config | purpose = HelpfullyDisabled })
```
Comment (tesk9) [inline src/Nri/Ui/Tooltip/V3.elm:842]: Maybe the `purpose` should be `AuxillaryDescription`? Or do we intend for the behavior to differ at any point between AuxillaryDescription and HelpfullyDisabled?

File: src/Nri/Ui/Tooltip/V3.elm [hunk 2]
```diff
@@ -835,6 +836,12 @@ auxiliaryDescription =
     Attribute (\config -> { config | purpose = AuxillaryDescription })
 
 
+{-| -}
```
Comment (tesk9) [inline src/Nri/Ui/Tooltip/V3.elm:839]: Could you please add some documentation here? 🙏

**Review by tesk9 — COMMENTED** [review]
Love the example change! Very clear guidance ❤️ 

Could you please add the helpfullyDisabled change to the Tooltip changelog? 🙏 

---

I'm noticing the tooltip opens on:
- focus
- mouse in

The tooltip closes on:
- esc
- mouse out

I'm noticing that nothing happens when I use space or enter -- the tooltip doesn't close and reopen. I expected it to open and close for me -- am I misremembering the behavior we wanted? 🤔

**Comment by charbelrami** [comment]
Thanks, Tessa!

> I'm noticing the tooltip opens on:
> 
> * focus
> * mouse in
> 
> The tooltip closes on:
> 
> * esc
> * mouse out
> 
> I'm noticing that nothing happens when I use space or enter -- the tooltip doesn't close and reopen. I expected it to open and close for me -- am I misremembering the behavior we wanted? 🤔

This is expected, as mentioned in the [Helpfully disabled components](https://paper.dropbox.com/doc/temporary-doc-Helpfully-disabled-components--B_5YGORe_Un6hamYU7qagW9MAg-2RUPgKnBsBNI7ScGDHS73#:uid=637996506239487824955939&h2=Our-helpfully-disabled-compone) doc:

> - They perform no action on click, neither from mouse onClick events nor from assistive technology click events (often activated by pressing SPACE or ENTER on the keyboard).
    - This “unclickability” is communicated visually via a dimmed-looking style and “do not enter” cursor icon on hover.
    - This “unclickability” is communicated semantically via the `aria-disabled="true"` property on the component.

**Review by BrianHicks — APPROVED** [review]
neato!

**Comment by BrianHicks** [comment]
oops, GIF:

cool-dog-gif-by-sarahmaes

**Pull request merged**